### ISSUE TITLE
Experiments: Table does not update after "dvc exp apply"

### ISSUE THREAD

**Issue by aschuh-hf** [iterative/vscode-dvc#4779]
Using the DVC 3.22.1 CLI command `dvc exp apply` in the Terminal did not trigger an update of the Experiments table of the VS Code extension v1.1.2 for the "workspace" row.

**Comment by aschuh-hf** [comment]
Possibly related issue: #4778 (i.e., table does not update after CLI commands modify something about the experiments)

**Comment by mattseddon** [comment]
What does the repository look like for this issue? Is it a monorepo? Is the `.git` folder within the workspace or outside? Is there a set of files that were updated by running `exp apply`? Were any of these params/metrics files, were they all data/deps files? What version of VS Code are you running? When you said "in the Terminal" did you mean the VS Code terminal or did you have a separate window open and ran the command in there?

**Comment by aschuh-hf** [comment]
> Is it a monorepo?

Yes.

> Is the .git folder within the workspace or outside?

The workspace points to a subdirectory of the Git repository.
The workspace root is the root of a DVC project (has a `.dvc/config`).

> Is there a set of files that were updated by running exp apply? Were any of these params/metrics files, were they all data/deps files?

I am using a `train` pipeline stage with deps and outs. The experiment output are model checkpoints and DVCLive metrics JSON and plots TSV files. The `dvc.lock` is being modified by `dvc exp apply` to update the hashes of these stage outs.

> What version of VS Code are you running?

```
Version: 1.82.2 (Universal)
Commit: abd2f3db4bdb28f9e95536dfa84d8479f1eb312d
Date: 2023-09-14T05:59:47.790Z (3 wks ago)
Electron: 25.8.1
ElectronBuildId: 23779380
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin x64 21.6.0
```

> When you said "in the Terminal" did you mean the VS Code terminal or did you have a separate window open and ran the command in there?

The integrated Terminal of VS Code.

**Comment by mattseddon** [comment]
Are you also experiencing https://github.com/iterative/dvc/issues/9860 in this repository?

**Comment by aschuh-hf** [comment]
> Are you also experiencing https://github.com/iterative/dvc/issues/9860 in this repository?

Yes, these issues are with the same Git repo and DVC project within this monorepo.

**Comment by mattseddon** [comment]
The filesystem watcher relies on the output from `exp show` to know which files to watch for changes. If the path(s) to these files are broken then the watcher won't be able to pick up changes.

**Comment by aschuh-hf** [comment]
Ah, I see. That could be a plausible explanation. Thanks for looking into it!

I did also run `dvc exp rename` and `dvc exp apply` for experiments that didn't have the issue of incorrect metrics / params file paths, however, and the `dvc.lock` file is the same in either way?

I don't remember now if I observed the non-update of the table also with those experiments that were not affected by this other DVC issue of incorrectly saved / returned paths by `dvc exp show`. The other experiments I manually re-placed were ones where the final experiment outputs were not properly synced from S3 after the `train` stage command finished.

For those I did:
```
dvc exp rename <exp> <exp>-needs-fixup
dvc exp apply <exp>-needs-fixup
aws s3 sync ...
dvc exp save -n <exp>
dvc exp remove <exp>-needs-fixup
```

I since got another experiment that was saved with not the latest, but second to last, outputs which I will patch up using the above commands. I can pay attention whether this will update the Experiments table, i.e., trigger the file watchers or not. I'll let you know what I find.
